packages/ra-ui-materialui/src/input/ReferenceInput.tsx

```diff
@@ -226,14 +242,11 @@ export const ReferenceInputView = ({
         isRequired,
         label,
         resource,
-        meta: {
-            ...meta,
-            helperText: warning || false,
-        },
+        helperText: warning || helperText,
```

Comment:
is it a bug fix or a breaking change?

Reply:
A breaking change we already discussed. All inputs should accept an helperText props and it should not come from the meta. Missing upgrade guide though

Reply:
However, I wonder whether it makes sense for the ReferenceXXX inputs as they receive another input as their child. I think this child should be one handling the helper text

Reply:
indeed

Reply:
By the way, the ReferenceXXX inputs used to override their child `helperText` to display a warning when a referenced item is not found. I believe they should not override anything and instead either pass the warning explicitly (new warning prop for their child) or pass it as the `error`.

Reply:
Can you add the upgrade guide for the helper text?

Reply:
I'd rather wait that we discuss about how to handle those cases